### PR TITLE
llvm: various improvements

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

1. ~~Enable bootstrap build on Linux~~. This fails with `gcc-5`. Let's try again when we switch to Ubuntu 18.04.
2. Clean up some duplication
3. Install the `lit` command line tool
4. Install Vim editor integration
5. Add patch to fix parallel builds on ARM

The `lit` tool adds very little to the bottle size, and it seems some
users might find it useful. Let's install it directly using Python to
avoid having to set `LLVM_INCLUDE_TESTS` to `ON`, as that has many other
side-effects that we don't need or want. This is also how Arch installs
`lit`. [1]

Closes #82479.

[1] https://github.com/archlinux/svntogit-packages/blob/5e7a8f09f53d0fb19c0680bb5b4691b1c6a78ad1/trunk/PKGBUILD#L79-L82

### CODE REVIEW

**Comment by carlocab** [comment]
On Linux:
```
CMake Error at /tmp/llvm-20210809-5953-1takjgm/llvm-project-12.0.1.src/compiler-rt/lib/crt/CMakeLists.txt:71 (message):
  gcc-5: error: unrecognized command line option ‘-Wimplicit-fallthrough’

  gcc-5: error: unrecognized command line option
  ‘-Wcovered-switch-default’

  gcc-5: error: unrecognized command line option ‘-Wstring-conversion’

Call Stack (most recent call first):
  /tmp/llvm-20210809-5953-1takjgm/llvm-project-12.0.1.src/compiler-rt/lib/crt/CMakeLists.txt:92 (check_cxx_section_exists)
```
Enabling the bootstrap on Linux might have to wait until we switch to Ubuntu 18.04.

File: Formula/llvm.rb [hunk 1]
```diff
@@ -114,21 +114,27 @@ def install
...
     end
 
-    sdk = MacOS.sdk_path_if_needed
+    # gcc-5 fails at building compiler-rt. Enable PGO
+    # build on Linux when we switch to Ubuntu 18.04.
+    pgo_build = false
     on_macos do
       args << "-DLLVM_BUILD_LLVM_C_DYLIB=ON"
       args << "-DLLVM_ENABLE_LIBCXX=ON"
       args << "-DLLVM_CREATE_XCODE_TOOLCHAIN=#{MacOS::Xcode.installed? ? "ON" : "OFF"}"
```
Comment (SMillerDev) [inline Formula/llvm.rb:132]: This will always be `ON` on CI I think

File: Formula/llvm.rb [hunk 2]
```diff
@@ -114,21 +114,27 @@ def install
       -DCLANG_VENDOR_UTI=org.#{tap.user.downcase}.clang
     ]
 
+    sdk = MacOS.sdk_path_if_needed
```
Comment (SMillerDev) [inline Formula/llvm.rb:117]:
```suggestion
    macos_sdk = MacOS.sdk_path_if_needed
```

File: Formula/llvm.rb (hunk 1 above)
Comment (carlocab) [reply Formula/llvm.rb:140]: Do you mean `LLVM_CREATE_XCODE_TOOLCHAIN`? That's true. I'll move it out of the `on_macos` block so we don't need to declare it in two places.

**Comment by danielnachun** [comment]
> On Linux:
> 
> ```
> CMake Error at /tmp/llvm-20210809-5953-1takjgm/llvm-project-12.0.1.src/compiler-rt/lib/crt/CMakeLists.txt:71 (message):
>   gcc-5: error: unrecognized command line option ‘-Wimplicit-fallthrough’
> 
>   gcc-5: error: unrecognized command line option
>   ‘-Wcovered-switch-default’
> 
>   gcc-5: error: unrecognized command line option ‘-Wstring-conversion’
> 
> Call Stack (most recent call first):
>   /tmp/llvm-20210809-5953-1takjgm/llvm-project-12.0.1.src/compiler-rt/lib/crt/CMakeLists.txt:92 (check_cxx_section_exists)
> ```
> 
> Enabling the bootstrap on Linux might have to wait until we switch to Ubuntu 18.04.

I don't want to waste too much time hunting this down right now, but I'm puzzled as to why this fails when we are actually building `compiler-rt` with `gcc-5` at some point.  If there was a way to drop these flags from being added, then it would probably work.  But we don't have to hold up progress on the rest of this just to get this figured out.

**Comment by carlocab** [comment]
That's not true -- we've never built `compiler-rt` with `gcc-5`. Recall that we pass `compiler-rt` in `LLVM_ENABLE_RUNTIMES` and not `LLVM_ENABLE_PROJECTS`.

**Comment by carlocab** [comment]
Linux runner timed out at testing dependents. This probably needs to be run on the self-hosted runner now.

**Comment by carlocab** [comment]
I think it'll be good to wait until Homebrew/brew#10769 is merged for this, as it'll be useful to rebuild `llvm` when we change the Big Sur SDK path.

Also, I should probably write a proper test for `lit`, as I think this formula being keg-only could make things a bit weird.

**Comment by carlocab** [comment]
Incorporated into #84364.

**Pull request closed without merging**